src/lib.rs

```diff
@@ -62,69 +106,466 @@ fn validate(payload: &[u8]) -> CallResult {
     }
 }
 
+// verify all images and return a PodSpec with the images replaced with the digest which was used for the verification
+fn verify_all_images_in_pod(spec: &PodSpec, signatures: &[Signature]) -> Result<PodSpec, String> {
```

Comment:
Maybe we could change the return signature to be:

```rust
Result<Option<PodSpec>, String>
```
With that:

  * `Ok(None)` -> accept, without any change
  * `Ok(Some(pod_spec))` -> accept, but with mutation

Reply:
done in https://github.com/kubewarden/verify-image-signatures/pull/4/commits/6cf9839979d4e5e78642006d40e7c373a65f4418